### ISSUE TITLE
Is design_mask leaky?

### ISSUE THREAD

**Issue by agarrubio** [MattMcPartlon/AttnPacker#7]
Hi!
My design_mask has 0's for the first 100 residues, and 1's for the rest of the protein. Although at low level, I get changes of in the first 100 residues. Is this expected?
Here are the results of some runs. Asterisks  mark the differences. My code is below.
```
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                                                                                                     **  *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDAAEALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKELREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                                                                *                                    **  *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEELRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDAAEALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKELREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                             *                                     *                                 **  *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAHRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRKEGATQIAYRSDDWRDLKEAWKKGADILIVDATDAAEALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKELREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                                                                                                     *** *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDAALALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKELREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                                                                                                     **  *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDAAEALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKELREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                                                                                                     **  *** **  *   **  *    *** ** ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDAAEALELVKELRAEGAKKIAFRSDDYEVLARAYEAGADILVVDDADAEKALELVKKLREEGVKTIAFRSDDAEVLLEARKAGA
=========
[INFO] Initializing AttnPacker Model
DILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDDWRDLKEAWKKGA
                     *                                                         *                     **  *** **  *   **  *    *** *  ***     *  ** *** *** **  *  ***  *    *** *  * *  
DILIVDATDKDEAWKQVEQLRKEGATQIAYRSDDWRDLKEAWKKGADILIVDATDKDEAWKQVEQLRREGATQIAYRSDNWRDLKEAWKKGADILIVDATDAEEALELVKELRAEGAKKIAFRSDDYEVLAEAYEAGADILVVDDADAEKALELVKKLREEGVKTIAFRSDDAEILLEARKAGA
```

The whole script:
```
#!/home/alejandro/mambaforge/envs/attnpacker/bin/python

import os
import sys
import torch
sys.path.append("/home/alejandro/AttnPacker")
import protein_learning.common.protein_constants as pc
from protein_learning.common.data.data_types.protein import safe_load_sequence, Protein
from protein_learning.models.inference_utils import Inference, make_predicted_protein, default
from protein_learning.models.inference_utils import format_prediction


# input protein from pdb
pdb_path= "/home/alejandro/Dropbox/work/timP1-1/inputs/5bvlP1.pdb"
fasta_path=None
protein = Protein.FromPDBAndSeq(pdb_path=pdb_path, seq=fasta_path, atom_tys=pc.ALL_ATOMS, missing_seq=fasta_path is None,load_ss=False,)
inseq=protein.seq

# design_mask
design_mask = torch.ones(len(protein)).bool()
design_mask[:100] = False

RESOURCE_ROOT = "/free/work/AttnPacker/AttnPackerPlusRotPTM"
runner = Inference(RESOURCE_ROOT, use_design_variant = True, use_rotamer_conditioning=True)
model, device = runner.get_model(), runner.device
model_input = runner.load_example(None,seq_mask=design_mask,dihedral_mask=design_mask, protein=protein)

prediction = model(model_input)
prediction = format_prediction(model,model_input,prediction)

# Show changes between input and output sequences
ident = ''.join([' ' if a == b else '*' for a, b in zip(inseq, prediction['seq'])])
print(inseq)
print(ident)
print(prediction['seq'])
```

**Comment by MattMcPartlon** [comment]
Hi Alejandro,

During training, each residue type in the input sequence is corrupted with 5% probability. This means that the residue type is randomly changed to some other type, and the model has to predict which residues were changed. This is a pretty standard trick with masked language models... in your case, the model is essentially guessing that some of your input sequence is "corrupt" and modifying accordingly. you can replace the first 100 residues in the predicted sequence with the ground truth and decode those side chain coordinates by passing this modified sequence to the 'make_predicted_protein' function.

Let me know if you have any questions. Very happy to help.

**Comment by agarrubio** [comment]
Many thanks, Matthew!
That answers my doubts.
